modules/cnvkit/batch/main.nf

```diff
@@ -18,28 +18,51 @@ process CNVKIT_BATCH {
     tuple val(meta), path("*.cnn"), emit: cnn, optional: true
     tuple val(meta), path("*.cnr"), emit: cnr, optional: true
     tuple val(meta), path("*.cns"), emit: cns, optional: true
+    tuple val(meta), path("*.pdf"), emit: pdf, optional: true
+    tuple val(meta), path("*.png"), emit: png, optional: true
     path "versions.yml"           , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
-    def normal_args = normal ? "--normal $normal" : ""
-    def fasta_args = fasta ? "--fasta $fasta" : ""
+    // execute samtools only when cram files are input
+    // input pair is assumed to have same extension if both exist
+    def is_cram = tumor.Extension == "cram" ? true : false
+    def tumor_out = is_cram ? tumor.BaseName + ".bam" : "${tumor}"
+    // do not run samtools on normal samples in tumor_only
+    def normal_exists = normal ? true: false
+    // tumor_only mode does not need fasta & target
+    // instead it requires a pre-computed reference which is built from fasta & target
+    def (normal_out, normal_args, fasta_args, target_args) = ["", "", "", ""]
     def reference_args = reference ? "--reference $reference" : ""
 
-    def target_args = ""
+    if (normal_exists){
+        def normal_prefix = normal.BaseName
+        normal_out = is_cram ? "${normal_prefix}" + ".bam" : "${normal}"
+        normal_args = normal_prefix ? "--normal $normal_out" : ""
+        fasta_args = fasta ? "--fasta $fasta" : ""
```

Comment:
Is the fasta only needed for normal?

Reply:
The `fasta_args` are indeed only needed when normal samples are provided. In the tumor_only mode the `reference.cnn` will be provided and then options to construct a new reference (-f/--fasta) should not be used.

Reply:
Sure, i'll remove the `params` section and i'll add a description.